toolchain/parser/parser_state.h

```diff
@@ -5,55 +5,27 @@
 #ifndef CARBON_TOOLCHAIN_PARSER_PARSER_STATE_H_
 #define CARBON_TOOLCHAIN_PARSER_PARSER_STATE_H_
 
-#include <cstdint>
-#include <iterator>
-
-#include "common/ostream.h"
-#include "llvm/ADT/StringRef.h"
+#include "toolchain/common/enum_base.h"
 
 namespace Carbon {
 
-class ParserState {
-  // Note that this must be declared earlier in the class so that its type can
-  // be used, for example in the conversion operator.
-  enum class StateEnum : uint8_t {
-#define CARBON_PARSER_STATE(Name) Name,
+CARBON_DEFINE_RAW_ENUM_CLASS(ParserState, uint8_t){
```

Comment:
```suggestion
CARBON_DEFINE_RAW_ENUM_CLASS(ParserState, uint8_t) {
```

Does clang-format remove a space here even if you try adding one? I'm noticing this through the code, but not sure if it was incidental.

Reply:
It does. I think it has a hard-coded set of macros that avoid this for things like GoogleTest (where I learned this macro idiom). I need to read the clang-format source code to see if there is a word or something I can use in the macro name to get this to fit into an existing rule.